canonical-web-and-design/ubuntu.com
# Update data model; switch to postgres by default

This speeds up bulk PUTs to /security/cve.

However, it would still take about 4.5 minutes to add 6500ish records, and 1.5 minutes to update them with minimal changes. For this reason, I'm here limiting the number of records you can insert at once to 300, which should always complete in less than 10 seconds.

I've updated https://github.com/nottrobin/ubuntu-cve-tracker/pull/2 to submit the CVEs in chunks of 300, which should still complete in under 5 minutes.

- Remove `reference` and `bug` tables, in favor of simply making them Arrays on `cve`s and `notice`s
- Change the primary key on tables for faster `get`s
- Make `package`s proper unique objects by extending `package_status` to include the `cve_id`, instead of the `package` table itself
- Update CVE and notice pages accordingly

QA
--

Run DB:

```
docker-compose rm -fs
docker-compose up
```

In another terminal:

```
dotrun --env DATABASE_URL=postgres://postgres:pw@localhost:5432/postgres exec alembic upgrade head
dotrun --env DATABASE_URL=postgres://postgres:pw@localhost:5432/postgres exec python3 scripts/create-releases.py
dotrun --env DATABASE_URL=postgres://postgres:pw@localhost:5432/postgres serve
```

In yet another terminal:

```
git clone -b load-cve git@github.com:carkod/ubuntu-cve-tracker
cd ubuntu-cve-tracker/scripts
python3 -m venv .venv
source .venv/bin/activate
pip3 install requests cvss
time python3 upload-cve.py ../active
```

Authenticate and then see the CVEs import. See the time is hopefully less than 5 minutes depending on how quickly you got through the auth step).

Also test importing USNs:

```
wget https://people.canonical.com/~ubuntu-security/usn/database.json
dotrun exec scripts/create-usns.py database.json
```

Go and browse through USNs and CVEs, check they look okay.
